plugins/modules/ec2_vpc_route_table.py

```diff
@@ -29,18 +34,23 @@
     description: Enable route propagation from virtual gateways specified by ID.
     type: list
     elements: str
+  purge_gw:
+    description: Purge existing gateway(s) from route table if different than gateway_id.
```

Comment:
Something like this for the description makes sense for you @jatorcasso? @tremble @markuman Does it make sense to you this kind of behaviour?
```suggestion
    description:
    - If I(purge_gw=True) and  I(gateway_id) is provided, the existing gateway is disassociated from the route table and the new I(gateway_id) is associated.
    - If I(purge_gw=True) and none I(gateway_id) is provided, the existing gateway is disassociated from the route table.
    - If I(purge_gw=False) and I(gateway_id) is provided, the I(gateway_id) is associated with the route table only if there is not one already associated.
```

Reply:
This is tricky, the module should define only one parameter to deal with gateway and with a special value like ``null`` to express that the gateway should be removed

Reply:
We can also have an easier bahavior and remove purge_gw:
```
 - If I(gateway_id) is provided, the existing gateway (if there is one already associated) is disassociated from the route table and the new I(gateway_id) is associated.
 -  If none  I(gateway_id) is provided, the existing configuration is retained.
```
Just wondering which is the one that makes more sense to have.

Reply:
Or to disassociate the existing gw we can specify something like an empty string `gateway_id: ''`, instead?

Reply:
@abikouo ah saw now you’re suggestion.

Reply:
I’m open to either way. I like having to explicitly set `purge_igw: yes` vs the alternative since that’s how subnets are being removed from a subnet route table.

Reply:
@abikouo @alinabuzachis `ec2_vol` handles it in a similar way to what @abikouo mentioned:

Reply:
Having an optional parameter default to being destructive feels dangerous. Is it possible that this change may end up breaking existing route tables?

Reply:
@gravesm after this discussion I removed the purge_gw parameter! That was outdated code

Reply:
Right, but since the `gateway_id` param is optional, if it's not provided this will, by default, disassociate a gateway. I guess what I'm asking is, if someone upgrades the collection, is this change going to potentially disassociate a gateway that they did not intend to disassociate?

Reply:
As @gravesm suggested, @jatorcasso you should do something like this (https://github.com/ansible-collections/amazon.aws/pull/690#discussion_r817567839) and you can disassociate https://github.com/ansible-collections/amazon.aws/pull/690#discussion_r817578247

Reply:
Oh, got it now. No, it doesn't do anything when it's not passed in - only when explicitly passed in with the value 'None'. I added integration tests to validate give them a look. Maybe I overlooked something.

Reply:
@jatorcasso Ok, thanks. Missed the update. Seems sensible to me!